### PR TITLE
system/adb: Support reset to bootloader and recovery mode

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
Could reboot to the different state by:
adb reboot bootloader
adb reboot recovery

## Testing
Pass CI

### CODE REVIEW

File: system/adb/adb_main.c [hunk 1]
```diff
@@ -49,7 +49,18 @@ void adb_log_impl(FAR const char *func, int line, FAR const char *fmt, ...)
...
+    {
+      boardctl(BOARDIOC_RESET, CONFIG_ADBD_RESET_RECOVERY);
+    }
+  else if (strcmp(target, "bootloader") == 0)
+    {
+      boardctl(BOARDIOC_RESET, CONFIG_ADBD_RESET_BOOTLOADER);
+    }
+  else
+    {
+      boardctl(BOARDIOC_RESET, 0);
```
Comment (pkarashchenko) [inline system/adb/adb_main.c:62]: Should we `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` or keep `0`?

File: system/adb/Kconfig [hunk 2]
```diff
@@ -49,6 +49,16 @@ config ADBD_TOKEN_SIZE
 
 endif # ADBD_AUTHENTICATION
 
+if BOARDCTL_RESET
+config ADBD_RESET_RECOVERY
+	int "Reset argument for recovery"
+	default 1
```
Comment (pkarashchenko) [inline system/adb/Kconfig:55]: `1` is reserved for `EXIT_FAILURE`. Should we default `ADBD_RESET_RECOVERY` to `2` and `ADBD_RESET_BOOTLOADER` to `3`?

Comment (xiaoxiang781216) [reply system/adb/Kconfig:55]: Th argument pass to BOARDCTL_RESET is used to control the reboot target, not the error code, I think.

File: system/adb/adb_main.c (hunk 1 above)
Comment (pkarashchenko) [reply system/adb/adb_main.c:62]: I just see calls `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` all over the apps code. Maybe we should modify it to `boardctl(BOARDIOC_RESET, 0);` in other places in next PR? What do you think?

Comment (pkarashchenko) [reply system/adb/adb_main.c:62]: I just see calls `boardctl(BOARDIOC_RESET, EXIT_SUCCESS);` all over the apps code. Maybe we should modify it to `boardctl(BOARDIOC_RESET, 0);` in other places in next PR? What do you think?

Comment (xiaoxiang781216) [reply system/adb/adb_main.c:67]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**